### PR TITLE
[Display_02] Built DisplayComm2WayInterfaceImplementationTests.java

### LINKED ISSUE #68: [Display_02] Build and Test Display_02 Production Classes
**What is the use case? Please describe.**
For this sprint task, you'll be building and testing all of the classes in the `edu.oakland.production.display02` package. Note: This issue is similar to #63 , but is tracked differently as helper and production classes have different deliverable dates.

**Sprint Task Setup**

To be done by @jcortes18 , @sydneyhill3901 , or @Volter43 before the rest of the team starts working on it:

1. Edit this issue message to add a checkbox for each class, and assign it to the correct team member from #57 .

**Prerequisites**

Please makes sure that your test plan from #57 for the class you are writing is closed and merged to main before starting this, to ensure that all of the documentation you need for this is in the correct spot.

**Steps**

To be done by each team member for each class they are developing a test plan for:

1. Each team member should watch the last ~18 minutes of @andrewdimmer 's [Test Driven Development Crash Course](https://youtu.be/eSc0aY4GhPs)
2. One person should create a branch off of main called `define-Name_Of_Class` (ex. the `Demo` classes and tests get build on a branch called `build-Demo`). Everyone else in the programming pair or group should then checkout that branch.
3. Using the information in the video, use test driven development to build all the required classes, stubs, and tests. Remember: you've already defined what you need to build at https://github.com/andrewdimmer/csi-3370-software-project/tree/main/docs/uml-class-diagrams/display02 !
4. Commit your changes with the message `[Display_02] Built Name_Of_Class`
5. Push your changes
6. Create a pull request with the name `[Display_02] Built Name_Of_Class` and the rest of the information like description, tags, etc. For reference, the settings on the right hand side should match this issue, with the exception that the assignees on the PR are just those who are pushing code.
7. Once your code is complete, add @andrewdimmer as a reviewer.

**Tasks to Complete**

- [x] Allocate Team Members to Classes - @jcortes18 , @sydneyhill3901 , or @Volter43 
- [x] Build DisplayCommInterfaceImplementationTests - @Dbaker19972 and @phardwoodward
- [ ] Build DisplayComm2WayInterfaceImplementationTests - @jmccloud827 and @Volter43
- [x] Build DisplayCommManagerImplementationTests - @MetalMan1992,  @Ahartman3 , and @yousifsallan 

**Additional context**
Note: if for some reason you need to change a public interface or shared data type, reach out to @andrewdimmer who can help you manage making that BREAKING CHANGE.

Note: if you run into merge conflicts, let @andrewdimmer know and he can help you resolve them.

### PR DESCRIPTION
**Pull request by jmccloud827**
Working on #68

### CODE REVIEW

**Review by sydneyhill3901 — CHANGES_REQUESTED** [review]
Try to address the errors and style issues listed, if you need help I will be available to work on this later today.

File: src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import edu.oakland.helper.admin.TrackData;
+
+public class DisplayComm2WayInterfaceImplementation implements DisplayComm2WayInterface {
+  DisplayCommManagerImplementation commManager;
+  
+  /**
+  * 
+  *
+  * @param DisplayCommManagerImplementation: The class that DisplayComm2Way will pass RFID to.
```
[the remark this answers is not in the document]
Comment (jcortes18) [reply src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java]: I think this should work once DisplayCommManagerImplementation is merged

**Comment by jcortes18** [comment]
Some errors in the test class. Please review, thanks!

**Review by Volter43 — APPROVED** [review]

File: src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  
+  /**
+  * Creates a DisplayComm2WayInterfaceImplementation to receive the rfid.
+  *
+  * @param displayCommManager The class that DisplayComm2Way will pass RFID to. 
+  *
+  */
+  public DisplayComm2WayInterfaceImplementation(DisplayCommManager displayCommManager) {
+    if (displayCommManager == null) {
+      throw new IllegalArgumentException("cannot be null");
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java]: Please make this more descriptive as to where the error is coming from.
```suggestion
      throw new IllegalArgumentException("displayCommManager cannot be null");

```

File: src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  
+  /**
+  * Passes the evaluated rfid along to DisplayCommManagerInterface.
+  *
+  * @return the TrackData
+  */
+  public TrackData passRfidRequest(int rfidNum) {
+
+    return commManager.evaluateRfidRequest(rfidNum);
+  }
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java]: Let's see if we can streamline this. I think we can remove all of this, correct? At the very least this needs to be private if you want to keep it.
```suggestion
```

File: src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  
+  
+  /**
+  * Determine the rfid request format.
+  *
+  * @return the TrackData
+  */
+  private TrackData determineUserRequestFormat(int rfidNum) {
+
+    return passRfidRequest(rfidNum);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display02/DisplayComm2WayInterfaceImplementation.java]: Let's see if we can streamline this. I think we can remove all of the private methods that are just pass throughs, right?
```suggestion
    return commManager.evaluateRfidRequest(rfidNum);

```

File: src/test/java/edu/oakland/test/display02/DisplayComm2WayInterfaceImplementationTests.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (andrewdimmer) [inline src/test/java/edu/oakland/test/display02/DisplayComm2WayInterfaceImplementationTests.java:2]: You need to make a method to test that the RFID method is passed in and a track data is passed, and checks that the status message of the TrackData is the output. That should finish your test coverage, and there are other examples of this merged into main if you look around in other Comm classes. Let me know if you have any other questions, and I'd be happy to help!

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Few recommendations to streamline your production code, and one missing test. Details are down below, and you should be able to find examples of the test in other test classes.

File: src/test/java/edu/oakland/test/display02/DisplayComm2WayInterfaceImplementationTests.java [hunk 6]
```diff
@@ -1,13 +1,26 @@
...
 
 import edu.oakland.production.display02.DisplayComm2WayInterfaceImplementation;
+import edu.oakland.test.display02.DisplayCommManagerStub;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 @DisplayName("DisplayComm2WayInterfaceImplementation Unit Tests")
 
 public class DisplayComm2WayInterfaceImplementationTests {
+  @Test
```
Comment (sydneyhill3901) [inline src/test/java/edu/oakland/test/display02/DisplayComm2WayInterfaceImplementationTests.java:12]: Glad you were able to figure out those coverage checks. Approving.

**Review by sydneyhill3901 — APPROVED** [review]
Looks good to me.

**Review by andrewdimmer — APPROVED** [review]
Looks good to me! You're all set to merge!

**Pull request merged**